src/commands/classified/external.rs

```diff
@@ -209,8 +209,12 @@ async fn run_with_stdin(
 
             let process_args = args.iter().map(|arg| {
                 let arg = expand_tilde(arg.deref(), || home_dir.as_ref());
-                if let Some(unquoted) = remove_quotes(&arg) {
-                    unquoted.to_string()
+                if argument_contains_whitespace(&arg) && argument_is_quoted(&arg) {
+                    if let Some(unquoted) = remove_quotes(&arg) {
+                        format!("'{}'", unquoted)
```

Comment:
I realized this won't work if the pass in a string with a single quote like "Sam's PR"